src/client.rs

```diff
@@ -280,6 +324,32 @@ impl Client {
         return self.get_connection(key).get()?.set(key, value, expiration);
     }
 
+    /// Set a key with associate value into memcached server with expiration seconds. The value will be serialized as JSON. This function is only available when the `json` feature is enabled.
+    ///
+    /// Example:
+    /// ```rust
+    /// use serde::Serialize;
+    ///
+    /// #[derive(Serialize)]
+    /// struct MyStruct {
+    ///     value: String,
+    ///     number: i32,
+    /// }
+    ///
+    /// let client = memcache::Client::connect("memcache://localhost:12345").unwrap();
+    /// let value = MyStruct {
+    ///     value: "hello".to_string(),
+    ///     number: 42,
+    /// };
+    /// client.set_json("foo", value, 10).unwrap();
+    /// client.flush().unwrap();
+    /// ```
+    #[cfg(feature = "json")]
+    pub fn set_json<V: Serialize>(&self, key: &str, value: V, expiration: u32) -> Result<(), MemcacheError> {
+        let value = serde_json::to_string(&value).map_err(|e| MemcacheError::JsonError(e))?;
```

Comment:
By doing it this way we can leverage the existing `ToMemcacheValue<Stream>` impl on `String` for storage. That'll make this easy to use as consumers won't have to worry about implementing the trait for their structs.